spec/unit/lib/wifi_user/gateway/gov_notify_sms_spec.rb

```diff
@@ -0,0 +1,79 @@
+require 'securerandom'
+require 'notifications/client'
+
+describe WifiUser::Gateway::GovNotifySMS do
+  let(:api_key) { "dummy_key-00000000-0000-0000-0000-000000000000-00000000-0000-0000-0000-000000000000" }
+
+  # modify the actual request
+  let(:template_id) { SecureRandom.uuid }
+  let(:parameters) { {} }
+  let(:phone_number) { '' }
+
+  # modify the stub
+  let(:api_url) { 'https://api.notifications.service.gov.uk/v2/notifications/sms' }
+  let(:return_status) { 200 }
+  let(:return_body) { {} }
+
+  before do
+    stub_request(:post, api_url).to_return(status: return_status, body: return_body.to_json)
+  end
+
+  let(:subject) do
+    described_class.new(api_key)
+      .execute(phone_number, template_id, template_parameters: parameters)
+  end
+
+  it 'sends an SMS request' do
+    subject
+    assert_requested :post, api_url,
+      times: 1,
+      body: {
+        phone_number: phone_number,
+        template_id: template_id,
+        personalisation: parameters
+      }
+  end
+
+  context 'on Success' do
+    it { expect(subject.success).to be true }
+  end
+
+  # These contexts are to test the different specified errors from the API
+  # https://docs.notifications.service.gov.uk/ruby.html#error-codes
+
+  context 'with unexpected error' do
+    # This will cause a parse error, which we want to see as an exception
+    let(:return_body) { '' }
+    it { expect { subject }.to raise_error(Exception) }
```

Comment:
Can we pick out a more specific exception here?

Reply:
I guess the reason this tests exists is to demonstrate that we still want to propagate exceptions, and only deal with BadRequestError only.

Do you have any suggests for how better to demonstrate?

Reply:
I would explicitly test the type of exception I'm expecting to bubble out - I forget what this is called in Ruby, is that the `BadRequestError` in this case? - because I'd be worried about false positives on this test.

If I wanted to demonstrate it doesn't capture arbitrary errors, I'd probably just provide three or four examples - possibly throwing errors from a mocked collaborator. I don't feel like that's necessary here, though.

Reply:
ok, will remove.

This was done before I narrowed the scope of what errors would be caught, so definitely less important